### DISCUSSION TITLE
Custom Quantization

### DISCUSSION THREAD

**Discussion by anthony-lemurian** [intel/neural-compressor#1785] [Q&A] [answered]
Hello, Im just wondering if its possible to define a custom data type to do WOQ in this repo? Im following the MX branch to see how they add that data type, however i wonder if there is a more straightforward approach since im only after WOQ

**Comment by yiliu30** [comment, marked as answer]
Hey @anthony-lemurian, thanks for showing interest in our project!

For WOQ, the basic idea is **quantize** and **dequantize** the tensor (weight) to mimic the quantization error. The main function for this is [quant_tensor](https://github.com/intel/neural-compressor/blob/7c0b700cae28e1374253b3ad6d413560c12e8f5c/neural_compressor/torch/algorithms/weight_only/utility.py#L251), which takes a tensor (weight) and certain configurations to select the [qdq_weight_actor](https://github.com/intel/neural-compressor/blob/7c0b700cae28e1374253b3ad6d413560c12e8f5c/neural_compressor/torch/algorithms/weight_only/utility.py#L226).

Taking 4-bits as an example, [qdq_weight_asym](https://github.com/intel/neural-compressor/blob/7c0b700cae28e1374253b3ad6d413560c12e8f5c/neural_compressor/torch/algorithms/weight_only/utility.py#L178) applies asymmetrical quantization and dequantization to the provided weight.

Hope this can give you some insights to define new data type :)
